numpy/core/overrides.py

```diff
@@ -12,6 +12,26 @@
 ARRAY_FUNCTION_ENABLED = bool(
     int(os.environ.get('NUMPY_EXPERIMENTAL_ARRAY_FUNCTION', 1)))
 
+array_function_like_doc = (
+    """like : array_like
+        Reference object to allow the creation of arrays which are not
+        NumPy arrays. If an array-like passed in as ``like`` supports
+        the ``__array_function__`` protocol, the result will be defined
+        by it. In this case, it ensures the creation of an array object
+        compatible with that passed in via this argument.
+
+        .. note::
+            The ``like`` keyword is an experimental feature pending on
+            acceptance of :ref:`NEP 35 <NEP35>`."""
+)
+
+def set_array_function_like_doc(public_api):
+    if public_api.__doc__ is not None:
+        public_api.__doc__ = public_api.__doc__.replace(
+            "${ARRAY_FUNCTION_LIKE}",
+            array_function_like_doc,
+        )
```

Comment:
```suggestion
        )
    return public_api  # to allow decorator use
```

Reply:
Thanks @eric-wieser , I tried something much more complicated and failed to achieve this, I'm glad it's so simple! Maybe one day I should take a few Python decorator classes with you! :)

That is now done in https://github.com/numpy/numpy/pull/16935/commits/1fc1b60eadd26523ff620d6b0f9386bcfa5b2276 .